src/community/features-templating/features-templating-core/src/test/java/org/geoserver/featurestemplating/builders/visitors/SelectionTemplateVisitorTest.java

```diff
@@ -0,0 +1,126 @@
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Literal;
+import org.xml.sax.helpers.NamespaceSupport;
+
+public class SelectionTemplateVisitorTest {
+    private SimpleFeatureType simpleFeatureType;
+    private FileSystemResourceStore store;
+
+    @Before
+    public void setup() {
+        store =
+                new FileSystemResourceStore(
+                        new File(
+                                "src/test/resources/org/geoserver/featurestemplating/builders/visitors"));
+        SimpleFeatureTypeBuilder tb = new SimpleFeatureTypeBuilder();
+        tb.add("geometry", Geometry.class);
+        tb.add("one", String.class);
+        tb.add("b", String.class);
+        tb.add("c", String.class);
+        tb.setName("testFeatureType");
+        SimpleFeatureType schema = tb.buildFeatureType();
+        schema.getDescriptor("b").getUserData().put(JDBCDataStore.JDBC_NATIVE_TYPENAME, "json");
+        schema.getDescriptor("c").getUserData().put(JDBCDataStore.JDBC_NATIVE_TYPENAME, "json");
+        this.simpleFeatureType = schema;
+    }
+
+    @Test
+    public void testPropertySelectionVisitor() throws IOException {
+        PropertySelectionHandler propertySelectionHandler = handler();
+        RootBuilder root = getBuilderTree("testTemplate.json");
+        PropertySelectionVisitor visitor =
+                new PropertySelectionVisitor(propertySelectionHandler, simpleFeatureType);
+        root = (RootBuilder) root.accept(visitor, null);
+        List<TemplateBuilder> children = root.getChildren();
+        TemplateBuilder builder = children.get(0);
+        builder = builder.getChildren().get(0);
+        for (TemplateBuilder child : builder.getChildren()) {
+            AbstractTemplateBuilder abstractBuilder = (AbstractTemplateBuilder) child;
+            String key = abstractBuilder.getKey(null);
+            if ("b".equals(key)) {
+                assertTrue(child instanceof DynamicPropertySelection);
+            } else if ("three".equals(key)) {
+                TemplateBuilder dynamicInclude = abstractBuilder.getChildren().get(1);
+                assertTrue(dynamicInclude instanceof DynamicIncludeFlatBuilder);
+            }
+            assertFalse("a".equals(key));
+            assertFalse("two".equals(key));
+        }
+    }
+
+    @Test
+    public void testPropertySelection2() throws IOException {
+        PropertySelectionHandler propertySelectionHandler = handler();
+        RootBuilder root = getBuilderTree("testTemplateOvr.json");
+        PropertySelectionVisitor visitor =
+                new PropertySelectionVisitor(propertySelectionHandler, simpleFeatureType);
+        root = (RootBuilder) root.accept(visitor, null);
+        List<TemplateBuilder> children = root.getChildren();
+        TemplateBuilder builder = children.get(0);
+        builder = builder.getChildren().get(0).getChildren().get(0);
+        for (TemplateBuilder child : builder.getChildren()) {
```

Comment:
This for loop does nothing, it does not iterate, the children are empty....

Reply:
In general this test class is weak compared to the work done in the template classes (I get that STAC tests likely cover it more, but it should do some bits here as well).

Reply:
fixed